src/modules/vtol_att_control/vtol_att_control_main.cpp

```diff
@@ -427,9 +428,28 @@ VtolAttitudeControl::is_fixed_wing_requested()
 		to_fw = _transition_command == vehicle_status_s::VEHICLE_VTOL_STATE_FW;
 	}
 
+	if(_abort_transition) {
+		to_fw = false;
+	}
+
 	return to_fw;
 }
 
+/*
+ * Abort transition
+ */
+void
+VtolAttitudeControl::abort_transition()
+{
+	if(!_abort_transition) {
+		// TODO: add mavlink log message
+		_abort_transition = true;
```

Comment:
Because _abort_transition currently just sets to_fw it's a bit misleading, you could also just toggle the to_fw flag then it's generic, but:

You also need to consider the manual case. Like this it will revert to MC and the user cannot transition again. Canceling the mission can be implemented later of course but is separate.

In the AUTO case we need to:
- reset the abort flag when checked
- also reset _transition_command otherwise it will transition again
- toggle to_fw

In manual:
- toggle to_fw
- wait until the user flicks the switch back to the mode it's in now and then reset the abort flag

Also we should only check for aborting the transition when it actually is in a transition (otherwise it could trigger one)